OpenRA.Mods.Common/UtilityCommands/OutputActorMiniYamlCommand.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using OpenRA.FileSystem;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	[Desc("ACTOR-TYPE [PATH/TO/MAP-DIRECTORY]", "Display the merged MiniYaml tree for the given actor type.")]
+	public class OutputActorMiniYamlCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name { get { return "--actor-yaml"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return args.Length == 2 || args.Length == 3;
+		}
+
+		void IUtilityCommand.Run(ModData modData, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = modData;
+
+			var actorType = args[1];
+
+			Map map = null;
+			if (args.Length == 3)
+				map = new Map(modData, modData.ModFiles.OpenPackage(args[2]));
+
+			var fs = map ?? modData.DefaultFileSystem;
+
+			var tree = LoadYamlRules(fs, modData.Manifest, map.RuleDefinitions);
+
+			MiniYaml actorMiniYaml;
+			if (!tree.TryGetValue(actorType, out actorMiniYaml))
+			{
+				Console.WriteLine("Could not find actor '{0}'.", actorType);
+				Console.WriteLine("Found keys: " + tree.Keys.JoinWith(","));
+				Environment.Exit(1);
+			}
+
+			Console.WriteLine(args[1] + ":");
+			RecursivePrintMiniYamlNodes(actorMiniYaml.Nodes, 1);
+		}
+
+		void RecursivePrintMiniYamlNodes(List<MiniYamlNode> nodes, int indent)
+		{
+			foreach (var n in nodes)
+			{
+				Console.WriteLine(new string('\t', indent) + "{0}: {1}", n.Key, n.Value.Value);
+				RecursivePrintMiniYamlNodes(n.Value.Nodes, indent + 1);
+			}
+		}
+
+		Dictionary<string, MiniYaml> LoadYamlRules(IReadOnlyFileSystem fs, Manifest manifest, MiniYaml additionalMiniYaml)
```

Comment:
You don't really need a dictionary either - just change the TryGetValue with a FirstOrDefault(n => n.Key ==)